README.md

```diff
@@ -1,3 +1,286 @@
 # create-cloud-deploy-release
 
-This action creates a release for Cloud Deploy.
+The `create-cloud-deploy-release` GitHub Action creates a [Cloud Deploy][cd]
+[release][cd-release] to manage the deployment of an application to one or more
+[Google Kubernetes Engine (GKE)][gke] or [Cloud Run][cloud-run]
+[targets][cd-target].
+
+## Prerequisites
+
+-   This action requires Google Cloud credentials that are authorized to access
+    the secrets being requested. See [Authorization](#authorization) for more
+    information.
+
+-   This action runs using Node 16. If you are using self-hosted GitHub Actions
+    runners, you must use runner version
+    [2.285.0](https://github.com/actions/virtual-environments) or newer.
+
+-   This action depends on the existence of a [Cloud Deploy][cd]
+    [delivery pipeline][cd-pipeline] that is configured for the targets to which
+    the application will be deployed.
+
+## Usage
+
+```yaml
+jobs:
+  job_id:
+    # ...
+
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+
+    steps:
+    - uses: 'actions/checkout@v3'
+
+    - uses: 'google-github-actions/auth@v1'
+      with:
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+   - id: create-release
+     uses: 'google-github-actions/create-cloud-deploy-release@v1'
+      with:
+        name: 'demo-app-v1.2.3'
+        description: 'Add new functionality to demo-app'
+        delivery_pipeline: 'demo-app-pipeline'
+        region: 'us-central1'
+        images: 'demo-app=us-central1-docker.pkg.dev/demo-app-project/demo-app-repo/demo-app:v1.2.3'
+        source: 'demo-app'
+```
+
+## Inputs
+
+-   `name`: (Required) The name of the release.
+
+-   `delivery_pipeline` (Required): The delivery pipeline to use for the release.
```

Comment:
Where does this come from? Let's not assume users are familiar with Cloud Deploy - can we link to some docs?